apps/web/src/features/courses/pages/components/course-description-modal.tsx

```diff
@@ -1,222 +1,609 @@
 import { addSpaceToCourseId } from '../../../../utils/courses.ts';
-import { computeAssetCdnUrl } from '../../../../utils/index.ts';
+import { formatTime } from '../../../../utils/date.ts';
+import { computeAssetCdnUrl, trpc } from '../../../../utils/index.ts';
 import type { TRPCRouterOutput } from '../../../../utils/trpc.ts';
 
-type Course = NonNullable<TRPCRouterOutput['content']['getCourse']>;
-const { useGreater } = BreakPointHooks(breakpointsTailwind);
+const hexToBase64 = (hexstring: string) => {
+  return Buffer.from(hexstring, 'hex').toString('base64');
+};
+
+const getFormattedUnit = (amount: number, unit: string, floating = 2) => {
+  let prefix = '';
+  if (amount > 0 && amount < 0.01) {
+    amount = 0.01;
+    prefix = `< `;
+  }
+
+  if (unit === 'sats') {
+    return `${prefix}${amount} sats`;
+  }
+
+  return `${prefix}${Intl.NumberFormat(undefined, {
+    style: 'currency',
+    currency: unit,
+    currencyDisplay: 'narrowSymbol',
+    minimumFractionDigits: floating,
+    maximumFractionDigits: floating,
+  }).format(amount)}`;
+};
+
+const DEFAULT_CURRENCY = 'EUR';
+
+interface PaymentData {
+  id: string;
+  pr: string;
+  onChainAddr: string;
+  amount: number;
+  checkoutUrl: string;
+}
+
+interface WebSocketMessage {
+  settled: boolean;
+}
+
 interface CourseDescriptionModalProps {
   course: NonNullable<TRPCRouterOutput['content']['getCourse']>;
   satsPrice: number;
   isOpen: boolean;
   onClose: (isPaid?: boolean) => void;
-  onPay: (
-    satsPrice: number,
-    withPhysical: boolean,
-    part?: number,
-    chapter?: number,
-  ) => void;
 }
 
+const sideStyle: CSSProperties = {
+  position: 'absolute',
+  top: 0,
+  left: 0,
+  width: '50%',
+  height: '100%',
+};
+
+const borderStyle: CSSProperties = {
+  borderWidth: 1,
+  borderColor: 'rgba(255, 255, 255, 0.25)',
+  overflow: 'hidden',
+};
+
+const baseButtonStyle: CSSProperties = {
+  borderRadius: 7,
+  color: 'white',
+  fontSize: 14,
+  height: 42,
+  display: 'flex',
+  alignItems: 'center',
+  justifyContent: 'center',
+  width: '100%',
+};
+
+interface RowTextProps {
+  children: string;
+  color?: string;
+}
+
+const RowText = ({
+  children,
+  style,
+}: RowTextProps & { style?: CSSProperties }) => (
+  <span style={{ fontSize: 14, ...style }}>{children}</span>
+);
+
+const RowLabel = ({ children, color }: RowTextProps) => (
+  <RowText style={{ color: color || 'rgba(255, 255, 255, 0.64)' }}>
+    {children}
+  </RowText>
+);
+
+const RowValue = ({ children, color }: RowTextProps) => (
+  <RowText
+    style={{
+      color: color || 'rgba(255, 255, 255, 1)',
+      textAlign: 'end',
+      maxWidth: '70%',
+      wordWrap: 'break-word',
+    }}
+  >
+    {children}
+  </RowText>
+);
+
+const Gap = () => <div style={{ height: 26 }} />;
+
+interface VStackProps {
+  children: React.ReactNode[];
+  style?: CSSProperties;
+}
+
+export const VStack = ({ children, style }: PropsWithChildren<VStackProps>) => {
```

Comment:
Can't we just use tailwind ?
=> <div className="flex flex-col'>
Instead of creating a component for that ?

Reply:
The `VStack` component is used to insert a specific gap between an array of children.

But we can indeed replace
`<div style={{ display: 'flex', flexDirection: 'column', ...style }}>`
with 
<div className="flex flex-col" style={style}>